admin/failures/overview/endpoint.js

```diff
@@ -13,28 +13,24 @@ module.exports = (purchases) => (req, res) => {
         let failedOrders = orders
             .filter(x => x.status == "failed" || (x.data.errors && x.data.errors.length))
             .sort((a, b) => {
-                if(!a.data.errors || !a.data.errors.length) {
-                    return -1;
-                }
-                if(!b.data.errors || !b.data.errors.length) {
+                let aLatestTimestamp, bLatestTimestamp;
+
+                !a.data.errors || !a.data.errors.length ? aLatestTimestamp = a.data.viewModel.date : aLatestTimestamp = a.data.errors[a.data.errors.length - 1].at;
+                !b.data.errors || !b.data.errors.length ? bLatestTimestamp = b.data.viewModel.date : bLatestTimestamp = b.data.errors[b.data.errors.length - 1].at;

```

Comment:
This is some of the most difficult-to-read code I've ever seen :smile:

Keep it simple: do simple if-statements, extract variables, etc. instead of this. Readability is more important than almost anything else!

Reply:
should be fixed now